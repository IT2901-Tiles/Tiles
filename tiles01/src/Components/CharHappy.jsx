import React from "react"
import Character from './Character'

function CharHappy() {
    return (
        <div className='CharHappy'>

            <svg viewBox="0 0 165 150">
                <Character/> 
                <g className="LightBulb">
                    <line id="lightStick1" x1="77" y1="4" x2="82" y2="9" stroke="black" strokeWidth="2">Â¨
                        <animate
                            attributeName = "stroke"
                            keyTimes="0;0.25;0.5;0.75;1"
                            values="black;yellow;yellow;black;black"
                            dur="0.75s"                        
                            repeatCount="indefinite" 
                        />
                    </line>
                    <line id="lightStick2" x1="83" y1="2" x2="85" y2="8" stroke="yellow" strokeWidth="2">
                        <animate
                            attributeName = "stroke"
                            keyTimes="0;0.25;0.5;0.75;1"
                            values="yellow;yellow;yellow;black;black"
                            dur="0.75s"                        
                            repeatCount="indefinite" 
                            />
                    </line>
                    <line id="lightStick3" x1="91" y1="2" x2="89" y2="8" stroke="black" strokeWidth="2">
                        <animate
                            attributeName = "stroke"
                            keyTimes="0;0.25;0.5;0.75;1"
                            values="black;yellow;yellow;black;black"
                            dur="0.75s"                        
                            repeatCount="indefinite"
                            />
                    </line>
                    <line id="lightStick4" x1="97" y1="4" x2="93" y2="9" stroke="yellow" strokeWidth="2">
                        <animate
                            attributeName = "stroke"
                            keyTimes="0;0.25;0.5;0.75;1"
                            values="yellow;yellow;yellow;black;black"
                            dur="0.75s"                        
                            repeatCount="indefinite" 
                        />
                    </line>
                    <circle id="lightBulb" cx="87.5" cy="15" r="5" fill="yellow" stroke="black">
                        <animate
                            attributeName = "fill"
                            keyTimes="0;0.25;0.5;0.75;1"
                            values="yellow;yellow;yellow;#ffcc00;#ffcc00"
                            dur="0.5s"                        
                            repeatCount="indefinite" 
                            />
                    </circle>
                    <line id="lightStickMain" x1="87.5" y1="20" x2="87.5" y2="28" stroke="black" strokeWidth="2"/>

                </g>
                <g className="happyEyes"> 
                    <path id="leftEyeHappy" d="M 72 53 q 6 -11 11 0" stroke="black" stroke-width="3" fill="none"/>
                    <path id="rightEyeHappy" d="M 90 53 q 6 -11 11 0" stroke="black" stroke-width="3" fill="none"/>
                    <animateTransform
                        attributeName = "transform"
                        attributeType = "XML"
                        type = "translate"
                        keyTimes = "0; 0.5; 1"
                        values = "0 0; 0 -2; 0 0"
                        dur = "1s"
                        repeatCount = "indefinite"
                    />
                </g>
                
                <g className="leftArm" transform="rotate(40 53.5 86)"> 
                    <circle id="shoulderL" cx="53.5" cy="86" r="5" fill="grey" stroke="black" />
                    <rect id="overarmL" x="50" y="93" height="16" width="7" fill="grey" stroke="black"/>
                    <g className="leftArmUnder">
                        <circle id="elbowL" cx="53.5" cy="115.5" r="5" fill="grey" stroke="black" />
                        <rect id="underarmL" x="31" y="113" height="7" width="16" fill="grey" stroke="black" transform="rotate(20 53.5 115.5)"/>
                        <rect id="handLL" x="16" y="112" height="5" width="13" fill="grey" stroke="black" transform="rotate(10 53.5 115.5)"/>
                        <rect id="handLR" x="17" y="117" height="5" width="13" fill="grey" stroke="black" transform="rotate(35 53.5 115.5)"/>
                        <animateTransform
                        attributeName = "transform"
                        attributeType = "XML"
                        type = "rotate"
                        keyTimes = "0; 0.5; 1"
                        values = "5 53.5 115.5;30 53.5 115.5; 5 53.5 115.5"
                        dur = "1s"
                        repeatCount = "indefinite"
                    />
                    </g>
                </g>
                <g className="rightArm" transform="rotate(-40 121.5 86)">
                    <circle id="shoulderR" cx="121.5" cy="86" r="5" fill="grey" stroke="black" />           
                    <rect id="overarmR" x="118" y="93" height="16" width="7" fill="grey" stroke="black"/>
                    <g className="rightArmUnder">
                        <circle id="elbowR" cx="121.5" cy="115.5" r="5" fill="grey" stroke="black" />
                        <rect id="underarmR" x="128" y="113" height="7" width="16" fill="grey" stroke="black" transform="rotate(-20 121.5 115.5)"/>
                        <rect id="handRL" x="140" y="103" height="5" width="13" fill="grey" stroke="black" transform="rotate(-30 140 101)"/>
                        <rect id="handRR" x="145" y="107" height="5" width="13" fill="grey" stroke="black" transform="rotate(10 145 107)"/> 
                        <animateTransform
                            attributeName = "transform"
                            attributeType = "XML"
                            type = "rotate"
                            keyTimes = "0; 0.5; 1"
                            values = "-5 121.5 115.5;-30 121.5 115.5; -5 121.5 115.5"
                            dur = "1s"
                            repeatCount = "indefinite"
                        />
                    </g>
                </g>  
            </svg>
        </div>
    )
}

export default CharHappy;
