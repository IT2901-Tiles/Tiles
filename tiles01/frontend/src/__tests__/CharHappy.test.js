import {configure, shallow} from "enzyme"
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {cleanup} from '@testing-library/react';
import CharHappy from '../Components/CharHappy';
import renderer from 'react-test-renderer';


afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('CharHappy', () => {
    const wrapper = shallow(<CharHappy/>)
    it('renders correctly', () => {
        const tree = renderer.create(<CharHappy/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders Character', () => {
        expect(wrapper.find('Character')).toHaveLength(1);
    })
});