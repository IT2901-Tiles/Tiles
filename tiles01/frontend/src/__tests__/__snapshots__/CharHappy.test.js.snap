// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharHappy renders correctly 1`] = `
<div
  className="CharHappy"
>
  <svg
    className="svgCharacter"
    viewBox="0 0 165 150"
  >
    <g
      className="whole"
    >
      <g
        className="HeadParts"
      >
        <rect
          fill="grey"
          height="35"
          stroke="black"
          width="55"
          x="60"
          y="36"
        />
        <rect
          fill="grey"
          height="15"
          stroke="black"
          width="5"
          x="53"
          y="46"
        />
        <rect
          fill="grey"
          height="15"
          stroke="black"
          width="5"
          x="117"
          y="46"
        />
        <rect
          fill="grey"
          height="6"
          stroke="black"
          width="15"
          x="80"
          y="28"
        />
        <circle
          cx="87.5"
          cy="61"
          fill="red"
          r="8"
          stroke="black"
        />
        <rect
          fill="grey"
          height="10"
          width="19"
          x="78"
          y="52"
        />
        <line
          stroke="black"
          x1="79"
          x2="96"
          y1="62"
          y2="62"
        />
      </g>
      <g
        className="BodyParts"
      >
        <rect
          fill="grey"
          height="8"
          stroke="black"
          width="15"
          x="80"
          y="73"
        />
        <rect
          fill="grey"
          height="35"
          stroke="black"
          width="55"
          x="60"
          y="83"
        />
      </g>
    </g>
    <g
      className="LightBulb"
    >
      <line
        stroke="black"
        strokeWidth="2"
        x1="77"
        x2="82"
        y1="4"
        y2="9"
      >
        Â¨
        <animate
          attributeName="stroke"
          dur="0.75s"
          keyTimes="0;0.25;0.5;0.75;1"
          repeatCount="indefinite"
          values="black;yellow;yellow;black;black"
        />
      </line>
      <line
        stroke="yellow"
        strokeWidth="2"
        x1="83"
        x2="85"
        y1="2"
        y2="8"
      >
        <animate
          attributeName="stroke"
          dur="0.75s"
          keyTimes="0;0.25;0.5;0.75;1"
          repeatCount="indefinite"
          values="yellow;yellow;yellow;black;black"
        />
      </line>
      <line
        stroke="black"
        strokeWidth="2"
        x1="91"
        x2="89"
        y1="2"
        y2="8"
      >
        <animate
          attributeName="stroke"
          dur="0.75s"
          keyTimes="0;0.25;0.5;0.75;1"
          repeatCount="indefinite"
          values="black;yellow;yellow;black;black"
        />
      </line>
      <line
        stroke="yellow"
        strokeWidth="2"
        x1="97"
        x2="93"
        y1="4"
        y2="9"
      >
        <animate
          attributeName="stroke"
          dur="0.75s"
          keyTimes="0;0.25;0.5;0.75;1"
          repeatCount="indefinite"
          values="yellow;yellow;yellow;black;black"
        />
      </line>
      <circle
        cx="87.5"
        cy="15"
        fill="yellow"
        r="5"
        stroke="black"
      >
        <animate
          attributeName="fill"
          dur="0.5s"
          keyTimes="0;0.25;0.5;0.75;1"
          repeatCount="indefinite"
          values="yellow;yellow;yellow;#ffcc00;#ffcc00"
        />
      </circle>
      <line
        stroke="black"
        strokeWidth="2"
        x1="87.5"
        x2="87.5"
        y1="20"
        y2="28"
      />
    </g>
    <g
      className="happyEyes"
    >
      <path
        d="M 72 53 q 6 -11 11 0"
        fill="none"
        stroke="black"
        stroke-width="3"
      />
      <path
        d="M 90 53 q 6 -11 11 0"
        fill="none"
        stroke="black"
        stroke-width="3"
      />
      <animateTransform
        attributeName="transform"
        attributeType="XML"
        dur="1s"
        keyTimes="0; 0.5; 1"
        repeatCount="indefinite"
        type="translate"
        values="0 0; 0 -2; 0 0"
      />
    </g>
    <g
      className="leftArm"
      transform="rotate(40 53.5 86)"
    >
      <circle
        cx="53.5"
        cy="86"
        fill="grey"
        r="5"
        stroke="black"
      />
      <rect
        fill="grey"
        height="16"
        stroke="black"
        width="7"
        x="50"
        y="93"
      />
      <g
        className="leftArmUnder"
      >
        <circle
          cx="53.5"
          cy="115.5"
          fill="grey"
          r="5"
          stroke="black"
        />
        <rect
          fill="grey"
          height="7"
          stroke="black"
          transform="rotate(20 53.5 115.5)"
          width="16"
          x="31"
          y="113"
        />
        <rect
          fill="grey"
          height="5"
          stroke="black"
          transform="rotate(10 53.5 115.5)"
          width="13"
          x="16"
          y="112"
        />
        <rect
          fill="grey"
          height="5"
          stroke="black"
          transform="rotate(35 53.5 115.5)"
          width="13"
          x="17"
          y="117"
        />
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          dur="1s"
          keyTimes="0; 0.5; 1"
          repeatCount="indefinite"
          type="rotate"
          values="5 53.5 115.5;30 53.5 115.5; 5 53.5 115.5"
        />
      </g>
    </g>
    <g
      className="rightArm"
      transform="rotate(-40 121.5 86)"
    >
      <circle
        cx="121.5"
        cy="86"
        fill="grey"
        r="5"
        stroke="black"
      />
      <rect
        fill="grey"
        height="16"
        stroke="black"
        width="7"
        x="118"
        y="93"
      />
      <g
        className="rightArmUnder"
      >
        <circle
          cx="121.5"
          cy="115.5"
          fill="grey"
          r="5"
          stroke="black"
        />
        <rect
          fill="grey"
          height="7"
          stroke="black"
          transform="rotate(-20 121.5 115.5)"
          width="16"
          x="128"
          y="113"
        />
        <rect
          fill="grey"
          height="5"
          stroke="black"
          transform="rotate(-30 140 101)"
          width="13"
          x="140"
          y="103"
        />
        <rect
          fill="grey"
          height="5"
          stroke="black"
          transform="rotate(10 145 107)"
          width="13"
          x="145"
          y="107"
        />
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          dur="1s"
          keyTimes="0; 0.5; 1"
          repeatCount="indefinite"
          type="rotate"
          values="-5 121.5 115.5;-30 121.5 115.5; -5 121.5 115.5"
        />
      </g>
    </g>
  </svg>
</div>
`;
